fetchcode/giturl.py

```diff
@@ -0,0 +1,74 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+
+from urllib.parse import urlparse
+from pathlib import Path
+
+
+def can_handle(source):
+    """
+	Take source's URL as input and then check it's scheme
+	if it not exits in any of http, https or git then it's 
+	not a valid VCS URL to be installed
+	"""
+    url_parts = urlparse(source)
+    if url_parts.scheme in ('https', 'git', 'git+ssh', 'ssh', 'git+https'):
+	    return True
+
+
+def clone(source, dest, branch='master', depth=None):
+	"""
+	Takes source, branch name, depth and destination as input
+	and check if the source URL can be handled and if destination doesn't already exists
+	If it doesn't exist make directory for VCS URL at that destination and clone the repo at
+	that destination
+	"""
+	if not can_handle(source):
+	    raise UnhandledSource('Failed to decode a Git/GitHub URL: {}'.format(source))
+
+	if os.path.exists(dest):
+	    print('Can not clone since repository already exists')
+	else:
+		os.mkdir(dest)
```

Comment:
This line is out of line with the rest of the file.

You need to indent in increments of 4 spaces per line of indentation. This line should be indented 8 spaces in this case.